core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -727,6 +735,45 @@ public static void toArmeria(io.netty.handler.codec.http.HttpHeaders inHeaders,
         if (cookieJoiner != null && cookieJoiner.length() != 0) {
             out.add(HttpHeaderNames.COOKIE, cookieJoiner.toString());
         }
+
+        maybeSetContentLengthUnknown(inHeaders.contains(HttpHeaderNames.CONTENT_LENGTH), out);
+    }
+
+    private static void maybeSetContentLengthUnknown(boolean hasContentLength, HttpHeadersBuilder out) {
+        if (hasContentLength) {
+            return;
+        }
+
+        HttpMethod method = null;
+        if (out instanceof RequestHeadersBuilder) {
+            method = ((RequestHeadersBuilder) out).method();
+        }
+
+        final boolean isContentAlwaysEmpty;
+        if (method != null) {
+            isContentAlwaysEmpty = isContentAlwaysEmpty(method);
+        } else {
+            isContentAlwaysEmpty = false;
+        }
+
+        if (!isContentAlwaysEmpty) {
+            // Set isContentLengthUnknown to true not to override the content-length when the HTTP objects are
+            // aggregated.
+            out.isContentLengthUnknown(true);
+        }
+    }
+
+    private static boolean isContentAlwaysEmpty(HttpMethod method) {
+        switch (method) {
+            case CONNECT:
+            case GET:
```

Comment:
Question) I understood that `GET` can have an empty request, but a non-empty response. This code path seems to be called by both `HttpRequestDecoder`, `HttpResponseDecoder`.

Why can't we just always set `isContentLengthUnknown=true` when parsing a request/response from remote if `content-length` does not exist in the headers? (not the trailers)

Reply:
> This code path seems to be called by both HttpRequestDecoder, HttpResponseDecoder.

I thought `method` will be null for `HttpResponseDecoder`.

> Why can't we just always set isContentLengthUnknown=true when parsing a request/response from remote if content-length does not exist in the headers?

I tried to fill `content-length` to 0 when we can be sure that the length of content is zero.